tools/clang-format/clang-format-diff.py

```diff
@@ -65,11 +66,24 @@ def main():
 
   # Extract changed lines for each file.
   filename = None
+  tmpfile = None
   lines_by_file = {}
   for line in sys.stdin:
     match = re.search('^\+\+\+\ (.*?/){%s}(\S*)' % args.p, line)
     if match:
+      # Do not use the local file but the version of the file that
+      # is committed in HEAD.
       filename = match.group(2)
+      command = ("git show HEAD:" + filename).split(" ")
```

Comment:
``HEAD`` feels weird here. What happened if your are not pushing the branch actually checkout ? Maybe we would like to have a optional argument to clang-format-diff set to ``HEAD`` by default.

Reply:
Agreed, I had not thought of this case. I'll add the argument.